### PR TITLE
NodeJS: Migrate Grpc Doc Gen to MVVM

### PR DESCRIPTION
**Pull request by landrito**
#### Still need to be completed.
- [x] Remove NodeJSGapicContext when NodeJS MVVM Surface Gen happens.

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/nodejs/NodeJSSurfaceNamer.java [hunk 1]
```diff
@@ -336,4 +341,66 @@ private String getParamTypeNoCardinality(ModelTypeTable typeTable, TypeRef type)
...
+        // Add a break if there was field docs and an extra description.
+        lines.add("");
+      }
+      lines.add(extraFieldDescription);
+    }
+    return lines.build();
+  }
+
+  private String getExtraFieldDescription(Field field) {
+    boolean fieldIsMap = field.getType().isMessage() && !field.getType().isMap();
```
Comment (eoogbe) [inline src/main/java/com/google/api/codegen/transformer/nodejs/NodeJSSurfaceNamer.java]: called `fieldIsMap`, but checks if `!isMap`. I think you want `fieldIsMessage`

Comment (landrito) [inline src/main/java/com/google/api/codegen/transformer/nodejs/NodeJSSurfaceNamer.java]: Good catch. Done!

File: src/test/java/com/google/api/codegen/testdata/nodejs_doc_message_library.baseline [hunk 2]
```diff
@@ -999,9 +1001,13 @@ var Book = {
    * @enum {number}
    */
   Rating: {
-    GOOD: 0,
+    /**
+     * GOOD enum description
+     */  GOOD: 0,
```
Comment (jmuk) [inline src/test/java/com/google/api/codegen/testdata/nodejs_doc_message_library.baseline]: Probably line breaking rule has been wrong (or at least weird) here, could you make a fix on the new template?  There should be a linebreak between the comment and the definition, like
```javascript
/**
 * GOOD enum description
 */
GOOD: 0,
```

Comment (landrito) [inline src/test/java/com/google/api/codegen/testdata/nodejs_doc_message_library.baseline]: Done!

**Comment by landrito** [comment]
Just removed the leftover files after the migration. PTAL.

File: src/main/java/com/google/api/codegen/viewmodel/GrpcMessageDocView.java [hunk 3]
```diff
@@ -21,12 +21,20 @@
...
+
   public abstract List<String> lines();
 
   public abstract List<ParamDocView> properties();
 
   public abstract List<GrpcElementDocView> elementDocs();
 
+  public abstract String fileUrl();
+
+  public boolean hasProperties() {
```
Comment (eoogbe) [inline src/main/java/com/google/api/codegen/viewmodel/GrpcMessageDocView.java]: Unnecessary method. Can just check `@if message.properties` in the snippet.

Comment (landrito) [inline src/main/java/com/google/api/codegen/viewmodel/GrpcMessageDocView.java]: Done!

**Pull request merged**